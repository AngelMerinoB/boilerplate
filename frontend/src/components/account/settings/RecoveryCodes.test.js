import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import RecoveryCodes from './RecoveryCodes';
Enzyme.configure({ adapter: new Adapter() });

const email = 'user@example.com';

describe('<RecoveryCodes />', () => {
  test('should render <RecoveryCodes /> without errors', () => {
    const wrapper = Enzyme.shallow(
      <RecoveryCodes email={email} codes={['00000000', '11111111']} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
