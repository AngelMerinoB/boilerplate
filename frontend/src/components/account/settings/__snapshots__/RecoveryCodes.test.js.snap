// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecoveryCodes /> should render <RecoveryCodes /> without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecoveryCodes
    codes={
        Array [
            "00000000",
            "11111111",
          ]
    }
    email="user@example.com"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="alert alert-primary text-left"
          role="alert"
>
          If you lose access to your authentication device, you can use one of these backup codes to login to your account. Each code may be used only once. Make a copy of these codes, and store it somewhere safe.
</div>,
        <p
          className="card-text text-center"
>
          <code>
                    00000000
                    <br />
          </code>
          <code>
                    11111111
                    <br />
          </code>
</p>,
        <p
          className="card-text text-right"
>
          <button
                    className="btn btn-primary"
                    onClick={[Function]}
          >
                    Print codes
          </button>
           
          <CopyToClipboard
                    onCopy={undefined}
                    options={undefined}
                    text="Boilerplate
          Two-Factor Authentication Backup Codes
          Email: user@example.com
          
          00000000
          11111111"
          >
                    <button
                              className="btn btn-primary"
                    >
                              Copy codes
                    </button>
          </CopyToClipboard>
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you lose access to your authentication device, you can use one of these backup codes to login to your account. Each code may be used only once. Make a copy of these codes, and store it somewhere safe.",
          "className": "alert alert-primary text-left",
          "role": "alert",
        },
        "ref": null,
        "rendered": "If you lose access to your authentication device, you can use one of these backup codes to login to your account. Each code may be used only once. Make a copy of these codes, and store it somewhere safe.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <code>
              00000000
              <br />
</code>,
            <code>
              11111111
              <br />
</code>,
          ],
          "className": "card-text text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "00000000",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "00000000",
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "00000000",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "code",
          },
          Object {
            "instance": null,
            "key": "11111111",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "11111111",
                <br />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "11111111",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "code",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn btn-primary"
              onClick={[Function]}
>
              Print codes
</button>,
            " ",
            <CopyToClipboard
              onCopy={undefined}
              options={undefined}
              text="Boilerplate
Two-Factor Authentication Backup Codes
Email: user@example.com

00000000
11111111"
>
              <button
                            className="btn btn-primary"
              >
                            Copy codes
              </button>
</CopyToClipboard>,
          ],
          "className": "card-text text-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Print codes",
              "className": "btn btn-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Print codes",
            "type": "button",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn btn-primary"
>
                Copy codes
</button>,
              "onCopy": undefined,
              "options": undefined,
              "text": "Boilerplate
Two-Factor Authentication Backup Codes
Email: user@example.com

00000000
11111111",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Copy codes",
                "className": "btn btn-primary",
              },
              "ref": null,
              "rendered": "Copy codes",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "p",
      },
    ],
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="alert alert-primary text-left"
            role="alert"
>
            If you lose access to your authentication device, you can use one of these backup codes to login to your account. Each code may be used only once. Make a copy of these codes, and store it somewhere safe.
</div>,
          <p
            className="card-text text-center"
>
            <code>
                        00000000
                        <br />
            </code>
            <code>
                        11111111
                        <br />
            </code>
</p>,
          <p
            className="card-text text-right"
>
            <button
                        className="btn btn-primary"
                        onClick={[Function]}
            >
                        Print codes
            </button>
             
            <CopyToClipboard
                        onCopy={undefined}
                        options={undefined}
                        text="Boilerplate
            Two-Factor Authentication Backup Codes
            Email: user@example.com
            
            00000000
            11111111"
            >
                        <button
                                    className="btn btn-primary"
                        >
                                    Copy codes
                        </button>
            </CopyToClipboard>
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you lose access to your authentication device, you can use one of these backup codes to login to your account. Each code may be used only once. Make a copy of these codes, and store it somewhere safe.",
            "className": "alert alert-primary text-left",
            "role": "alert",
          },
          "ref": null,
          "rendered": "If you lose access to your authentication device, you can use one of these backup codes to login to your account. Each code may be used only once. Make a copy of these codes, and store it somewhere safe.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <code>
                00000000
                <br />
</code>,
              <code>
                11111111
                <br />
</code>,
            ],
            "className": "card-text text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "00000000",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "00000000",
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "00000000",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "code",
            },
            Object {
              "instance": null,
              "key": "11111111",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "11111111",
                  <br />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "11111111",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "code",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn-primary"
                onClick={[Function]}
>
                Print codes
</button>,
              " ",
              <CopyToClipboard
                onCopy={undefined}
                options={undefined}
                text="Boilerplate
Two-Factor Authentication Backup Codes
Email: user@example.com

00000000
11111111"
>
                <button
                                className="btn btn-primary"
                >
                                Copy codes
                </button>
</CopyToClipboard>,
            ],
            "className": "card-text text-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Print codes",
                "className": "btn btn-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Print codes",
              "type": "button",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn btn-primary"
>
                  Copy codes
</button>,
                "onCopy": undefined,
                "options": undefined,
                "text": "Boilerplate
Two-Factor Authentication Backup Codes
Email: user@example.com

00000000
11111111",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Copy codes",
                  "className": "btn btn-primary",
                },
                "ref": null,
                "rendered": "Copy codes",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "p",
        },
      ],
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
